compiler_opt/rl/corpus.py

```diff
@@ -12,31 +12,145 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
-"""ModuleSpec definition and utility command line parsing functions."""
+"""Corpus and related concepts."""
+import abc
 import math
 import random
 import re
 
 from absl import logging
 from dataclasses import dataclass
-from typing import Any, Dict, Iterable, List, Optional, Tuple
+from typing import Any, Dict, List, Optional, Tuple
 
 import json
 import os
+import tensorflow as tf
 
 from compiler_opt.rl import constant
 
+# Alias to better self-document APIs. Represents a complete, ready to use
+# command line, where all the flags reference existing, local files.
+FullyQualifiedCmdLine = Tuple[str, ...]
+
+
+def _localize_cmdline(
```

Comment:
this function is doing more than localizing cmdline. From readability perspective, how about:
1. still keep the cmd pre-processing(add/delete/replace) in ModuleSpec
2. only takes care of flag localizing here to go over all flags to replace the path prefix.

the benefit is that all the cmd pre-processing code (cmd loading, add/delete/replace conflict detection and merging, actually doing the job) are in one place so that easier to test/debug

Reply:
I like the outcome - even testing is easier - I think the answer to my question above is "yes", see the latest patch.

Reply:
yeah, that's what i mean